invenio_records_rest/utils.py

```diff
@@ -40,6 +40,33 @@
     PIDUnregisteredRESTError
 
 
+@cached_property
```

Comment:
The question is if we then need `get_default_endpoint_for` function instead using the property e.g. `endpoint_for_pid_type` directly so the code will look: `current_records_rest.endpoint_for_pid_type[pid_type]`

(other possible namings: `default_endpoints`, `pid_type_redirects`, `endpoints_for`, ...)